api-gateway/Jenkinsfile

```diff
@@ -0,0 +1,111 @@
+pipeline {
+    agent any
+
+    parameters {
+        string(name: 'BRANCH_NAME', defaultValue: 'develop', description: 'Branch to build')
+    }
+
+    environment {
+        DOCKER_CREDENTIALS_ID = 'c1269293-12a7-4ae4-a1fe-b048736d5658'
+        SLACK_CHANNEL = '#pipeline-notifications'
+        SLACK_TOKEN_CREDENTIAL_ID = 'slack-token'
+        COMMITTER_NAME = ''
+        TRIGGERED_BY = ''
+    }
+
+    stages {
+        stage('Fetch Committer Info') {
+            steps {
+                script {
+                    def commitInfo = sh(script: 'git log -1 --pretty=%an,%ae', returnStdout: true).trim().split(',')
+                    COMMITTER_NAME = commitInfo[0]?.trim() ?: 'Unknown'
+
+                    def userCause = currentBuild.rawBuild.getCause(hudson.model.Cause$UserIdCause)
+                    if (userCause) {
+                        TRIGGERED_BY = "Triggered manually by: ${userCause.getUserName()}"
+                    } else {
+                        TRIGGERED_BY = "Triggered by: ${COMMITTER_NAME}"
+                    }
+                    echo TRIGGERED_BY
```

Comment:
We discussed that in this project the pipelines will not run automatically on commits, right? The only way to run the pipeline will be through manual jenkins trigger.

Reply:
Now, we can run the pipeline through manual triggers only.